python-backend/app/api/permits/permit_amendment/resources/permit_amendment.py

```diff
@@ -21,7 +44,89 @@ def get(self, permit_guid=None, permit_amendment_id=None):
             if permit:
                 permit_amendments = PermitAmendment.find_by_permit_id(permit.permit_id)
                 if permit_amendments:
-                    result = {'amendments': [x.json() for x in permit_amendments]}
-                    return result
+                    return [x.json() for x in permit_amendments]
 
         return self.create_error_payload(404, 'Permit amendment(s) not found'), 404
+
+    @api.doc(params={
+        'permit_amendment_guid': 'Permit amendment guid.',
+        'permit_guid': 'Permit GUID'
+    })
+    @requires_role_mine_create
+    def post(self, permit_guid=None, permit_amendment_guid=None):
+        if not permit_guid:
+            return self.create_error_payload(400, 'Permit_guid must be provided'), 400
+        if permit_amendment_guid:
+            return self.create_error_payload(400, 'unexpected permit_amendement_id'), 400
+
+        permit = Permit.find_by_permit_guid(permit_guid)
+        if not permit:
+            return self.create_error_payload(404, 'permit does not exist'), 404
+
+        data = self.parser.parse_args()
+
+        received_date = data.get('received_date')
+        issue_date = data.get('issue_date')
+        authorization_end_date = data.get('authorization_end_date')
+
+        new_pa = PermitAmendment.create(
+            permit,
+            received_date,
+            issue_date,
+            authorization_end_date,
+            self.get_create_update_dict(),
+            save=True)
+
+        return new_pa.json()
+
+    @api.doc(params={
+        'permit_amendment_guid': 'Permit amendment guid.',
+        'permit_guid': 'Permit GUID'
+    })
+    @requires_role_mine_create
+    def put(self, permit_guid=None, permit_amendment_guid=None):
+        if not permit_amendment_guid:
+            return self.create_error_payload(400, 'permit_amendment_id must be provided'), 400
+        pa = PermitAmendment.find_by_permit_amendment_guid(permit_amendment_guid)
+        if not pa:
+            return self.create_error_payload(404, 'permit amendment not found'), 404
+
+        data = self.parser.parse_args()
+        current_app.logger.info(f'updating {pa} with >> {data.keys()}')
+
+        if 'received_date' in data.keys():
```

Comment:
Quick test in my terminal shows that the `keys()` method is redundant?
```
>>> 'received_date' in { 'received_date': 1 }
True
>>> 'received_date' in { 'issue_date': 1 }
False
```

Reply:
Learned something new, thanks.